include/gamepad/controller.hpp

```diff
@@ -17,31 +18,152 @@ enum EventType {
     ON_PRESS,
     ON_LONG_PRESS,
     ON_RELEASE,
+    ON_SHORT_RELEASE,
 };
 
 class Button {
         friend class Controller;
     public:
+        /// Whether the button has just been pressed
         bool rising_edge = false;
+        /// Whether the button has just been released
         bool falling_edge = false;
+        /// Whether the button is currently held down
         bool is_pressed = false;
-        uint32_t last_press_time = pros::millis();
-        uint32_t last_release_time = last_press_time;
+        /// How long the button has been held down
         uint32_t time_held = 0;
+        /// How long the button has been released
         uint32_t time_released = 0;
+        /// How long the threshold should be for the longPress and shortRelease events
         uint32_t long_press_threshold = 500;
+        /**
+         * @brief Register a function to run when the button is pressed.
+         *
+         * @param listenerName The name of the listener, this must be a unique name
+         * @param func The function to run when the button is pressed, the function MUST NOT block
+         * @return true The listener was successfully registered
+         * @return false The listener was not successfully registered (there is already a listener with this name)
```

Comment:
pad returns with the example code

```suggestion
         * @return false The listener was not successfully registered (there is already a listener with this name)
         *
```